### ISSUE TITLE
Root folder for projectile, treemacs and lsp-mode

### ISSUE THREAD

**Issue by wztdream** [emacs-lsp/lsp-mode#1989]
Hi,
I have suffered an issue of _unresolved import_ [issue96](https://github.com/emacs-lsp/lsp-python-ms/issues/96), finally I found the root case is that the root folder of a project in a workspace that is recognized by lsp-mode is not the root folder of the treemacs. Following [lsp-mode manul](https://emacs-lsp.github.io/lsp-mode/page/keybindings/) we need to add the root folder to a workspace by `s-l F a`, this seems a separate definition of root folder with respect to a treemacs, and they all have the name of _the root folder for a project_. This is very misleading. I believe the [issue96](https://github.com/emacs-lsp/lsp-python-ms/issues/96) are partially cased by this misleading. And the function to check what is the root folder for a file in lsp-mode is not interactive `M: (lsp-workspace-root)`, so it is difficult for people to do trouble shooting.

In summary, my question is what is the different of the definition of _root folder_ for a project between lsp-mode, treemacs and projectile. Is it possible to make them consistent?

BTW, in the hydra key for treemacs we can add a project to a workspace by `C-c C-p a`, this also need specify the root folder, but this root folder is not recognized by lsp-mode. Also in treemacs there is no where to check the root folder of lsp-mode too. It is misleading. So I think it will be much more clear for people to use if we can make all these three root folder consistent, and better to be available in treemacs.

**Comment by wztdream** [comment]
I see your solution in lsp-treemacs [feature](https://github.com/emacs-lsp/lsp-treemacs/issues/17)
